tests/broker.js

```diff
@@ -0,0 +1,63 @@
+"use strict";
+
+require('es6-shim');
+
+var net = require('net');
+var expect = require('chai').expect;
+var makeTcpReceiver = require('../tools/makeTcpReceiver');
+var boot2dockerIp = require('../tools/boot2dockerIp.js');
+var PRIVATE = require('../PRIVATE.json');
+
+
+describe('init', function() {
+
+	var sensorSocket;
+	var tcpSocketSensorReceiver;
+
+	// simulate sensor
+	beforeEach(function(ready){
+
+		boot2dockerIp()
+		.then(function(host){
+			sensorSocket = net.connect({
+		        host: host,
+		        port: 5100
+		    });
+
+			sensorSocket.on('connect', function(){
+		        tcpSocketSensorReceiver = makeTcpReceiver(sensorSocket, "\n");
+		        ready();
+		    });
+		})
+		.catch(function(error){
+			console.log("Error determining the host");
+		});
+
+	});
+
+
+	it('broker should register unknown sensor if password ok', function (done) {
+
+		tcpSocketSensorReceiver.on('message', function(message) {
+
+    		// console.log("received tcp data: ", message);
+    		var args = message.slice(4);
+    		var argsplit = args.split(" ");
+    		expect(argsplit[0]).to.equal("init");
+    		expect(parseInt(argsplit[1])).to.be.a("number");
```

Comment:
Given you're doing parseInt, it will always be a number. You need to verify is it different from `NaN`
 (`!Number.isNaN(x)` or maybe the test assertion can test this directly)